dashboard-api/internal/platform/server/handler/studies/list_test.go

```diff
@@ -0,0 +1,129 @@
+package studies
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	studiesmanager "github.com/vlab-research/vlab/dashboard-api/internal"
+	"github.com/vlab-research/vlab/dashboard-api/internal/platform/storage/storagemocks"
+)
+
+func TestHandler_List(t *testing.T) {
+	t.Run("should return a 200 with the requested studies when the repository returns them", func(t *testing.T) {
+		defaultOffset := 0
+		defaultLimit := 20
+		studyRepository := new(storagemocks.StudyRepository)
+		study := studiesmanager.NewStudy("5372ca9c-9fcd-42d4-a596-d90792909917", "Example Study", "example-study", time.Date(2020, time.November, 10, 23, 0, 0, 0, time.UTC))
+		studyRepository.On("GetStudies", mock.Anything, defaultOffset, defaultLimit).Return([]studiesmanager.Study{study}, nil)
+
+		res := performRequest("/studies", studyRepository)
+
+		assert.Equal(t, http.StatusOK, res.StatusCode)
+		assert.Equal(t, "application/json; charset=utf-8", res.Header.Get("Content-Type"))
+		assert.Equal(t, "{\"data\":[{\"id\":\"5372ca9c-9fcd-42d4-a596-d90792909917\",\"name\":\"Example Study\",\"slug\":\"example-study\",\"createdAt\":1605049200000}],\"pagination\":{\"nextCursor\":null}}", res.body)
+	})
+
+	t.Run("should return a 400 when the cursor query parameter is invalid", func(t *testing.T) {
+		res := performRequest("/studies?cursor=invalid-cursor", new(storagemocks.StudyRepository))
+
+		assert.Equal(t, http.StatusBadRequest, res.StatusCode)
+		assert.Equal(t, "application/json; charset=utf-8", res.Header.Get("Content-Type"))
+		assert.Equal(t, "{\"error\":\"Key: 'paginationQueryParams.CursorInBase64url' Error:Field validation for 'CursorInBase64url' failed on the 'base64url' tag\"}", res.body)
+	})
+
+	t.Run("should return a 400 when the number query parameter is invalid", func(t *testing.T) {
+		res := performRequest("/studies?number=101", new(storagemocks.StudyRepository))
+
+		assert.Equal(t, http.StatusBadRequest, res.StatusCode)
+		assert.Equal(t, "application/json; charset=utf-8", res.Header.Get("Content-Type"))
+		assert.Equal(t, "{\"error\":\"Key: 'paginationQueryParams.Number' Error:Field validation for 'Number' failed on the 'lte' tag\"}", res.body)
+	})
+
+	t.Run("should return a 500 when the repository returns an error", func(t *testing.T) {
+		studyRepository := new(storagemocks.StudyRepository)
+		studyRepository.On("GetStudies", mock.Anything, mock.Anything, mock.Anything).Return([]studiesmanager.Study{}, errors.New("unexpected-error"))
+		res := performRequest("/studies", studyRepository)
+
+		assert.Equal(t, http.StatusInternalServerError, res.StatusCode)
+	})
+
+	t.Run("should return a valid nextCursor when number of studies returned for the current page is equal to the requested ones", func(t *testing.T) {
```

Comment:
Nice documentation explaining expected behavior at the edge case!

Reply:
Glad that you liked it!!